pilot/pkg/proxy/envoy/v1/discovery.go

```diff
@@ -481,6 +489,19 @@ func (ds *DiscoveryService) ClearCacheStats(_ *restful.Request, _ *restful.Respo
 // clearCache will clear all envoy caches. Called by service, instance and config handlers.
 // This will impact the performance, since envoy will need to recalculate.
 func (ds *DiscoveryService) clearCache() {
+	clearCacheMutex.Lock()
```

Comment:
no need for mutex, this is run on a single event queue

Reply:
Just feel safer with a mutex - with the other changes in multicluster and other optimizations 
we may have multiple callers.

Reply:
how does this mutex work with the cache lock? there are things reading from the cache in parallel, right?

Reply:
This mutex only syncs the clearCache calls. 
Additional mutexes or other sync may be needed - we'll discuss it when that work is done, for now
it's useful to stop assuming everything is in one thread... Is there any harm in adding a mutex to clearCache ?